### DISCUSSION TITLE
QT6 is the future

### DISCUSSION THREAD

**Discussion by collin80** [collin80/SavvyCAN#774] [General]
As many people have found, QT5 has been deprecated for years now and most distros have begun to base their packages on QT6. Because of this, it is becoming more difficult for people to compile SavvyCAN as it has been targeting QT5. 

However, this was not entirely unexpected. Work from both myself and others has been done to update SavvyCAN to compile and run on QT6. This code is found in the QT6WIP branch. This branch also has a CD build automatically generated as "QT6 Development Build" in the releases. Today I've merged all the changes from the master branch into QT6WIP.

The plan, going forward, is to move this branch over to master and replace the QT5 codebase. The code will then require QT6 to compile. QT6 was released over 3 years ago so I think it's really time to leave QT5 behind. 

So long as no large showstopping bugs are found, pretty soon this transition will take place and everything will need to work in QT6 to be merged into the code base. This is normally not very difficult to do.

If people could please test the QT6 development release, that'd be great. Hopefully at this point it pretty much works just like the QT5 version.

**Comment by Pepe424** [comment]
The latest version a954f2e won't even run on macOS due to a missing library
```
dyld[54570]: Library not loaded: @rpath/QtDBus.framework/Versions/A/QtDBus
  Referenced from: <8359F3D2-B8C1-37D7-BCA7-D0FA76832F63> /Volumes/SavvyCAN/SavvyCAN.app/Contents/Frameworks/QtGui.framework/Versions/A/QtGui
  Reason: tried: '/Volumes/SavvyCAN/SavvyCAN.app/Contents/Frameworks/QtGui.framework/Versions/A/../../../QtDBus.framework/Versions/A/QtDBus' (no such file), '/Volumes/SavvyCAN/SavvyCAN.app/Contents/Frameworks/QtWidgets.framework/Versions/A/../../../QtDBus.framework/Versions/A/QtDBus' (no such file), '/Volumes/SavvyCAN/SavvyCAN.app/Contents/Frameworks/QtPrintSupport.framework/Versions/A/../../../QtDBus.framework/Versions/A/QtDBus' (no such file), '/Volumes/SavvyCAN/SavvyCAN.app/Contents/Frameworks/QtDBus.framework/Versions/A/QtDBus' (no such file), '/usr/local/lib/QtDBus.framework/Versions/A/QtDBus' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/lib/QtDBus.framework/Versions/A/QtDBus' (no such file), '/Volumes/SavvyCAN/SavvyCAN.app/Contents/Frameworks/QtDBus.framework/Versions/A/QtDBus' (no such file), '/usr/local/lib/QtDBus.framework/Versions/A/QtDBus' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/lib/QtDBus.framework/Versions/A/QtDBus' (no such file), '/Library/Frameworks/QtDBus.framework/Versions/A/QtDBus' (no such file), '/System/Library/Frameworks/QtDBus.framework/Versions/A/QtDBus' (no such file, not in dyld cache)
zsh: abort      ./Contents/MacOS/SavvyCAN
```

**Reply by Pepe424** [reply]
Version 5779460 is working but it is missing libraries for canbus (macOS arm64).

**Comment by collin80** [comment]
OK, I have a MacBook so I tried it there and, sure enough, it bombs with the same error you got. I have fixed this by explicitly saying it depends on DBus even though I have no idea at all why it would. I don't actually use DBus anywhere myself. Maybe one of the other parts of QT implicitly uses it somewhere. At any rate, causing it to be added makes it work. It still has no SerialBus drivers while there should be around 4 of them. So, that's the next problem to fix. MacOS has always been the third class citizen when it comes to CAN but at least some drivers should exist. I'll try to get them included.
